crowbar_framework/lib/execute.rb

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright 2013-2016, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "open3"
+
+class Execute
+  class << self
+    def bash(user, *params)
+      Rails.logger.debug("Execute bash \"#{params.join(", ")}\" as #{user}.")
+
+      if user.nil?
+        return bash_as(false, *params)
+      elsif user == "root"
+        return bash_as(true, *params)
+      else
+        execution_uid, execution_gid = uid_gid(user)
+        return false unless execution_uid
+
+        Process.fork do
+          Process.egid = execution_gid
+          Process.euid = execution_uid
```

Comment:
I don't understand how this can work: we're running as the `crowbar` user. This user cannot change it's euid (afaik). Don't we need to go through sudo/su/something else for that?